packages/loader/container-loader/src/container.ts

```diff
@@ -1167,20 +1193,19 @@ export class Container
 					);
 				}
 				assert(
-					this.attachState === AttachState.Attached,
+					this.attachmentData.state === AttachState.Attached,
```

Comment:
nit (preference): this line can stay the same

Reply:
The reason it needs to change is that it now narrows the type of this.attachmentData, which lets us access this.attachmentData.snapshot below without additional casting.